WordPress/Tests/Utils/I18nTextDomainFixerUnitTest.3.inc

```diff
@@ -0,0 +1,119 @@
+ *
+ * bbPress is forum software with a twist from the creators of WordPress.
+ *
+ * $Id: bbpress.php 6699 2017-09-13 21:35:09Z johnjamesjacoby $
+ *
+ * @package bbPress
+ * @subpackage Main
+ */
+
+/**
+ * Plugin Name: Text domain which should be fixed, docblock format.
+ * Plugin URI:  https://bbpress.org
+ * Description: bbPress is forum software with a twist from the creators of WordPress.
+ * Author:      The bbPress Contributors
+ * Author URI:  https://bbpress.org
+ * Version:     2.6-rc-5
+ * Text Domain: bbpress
+ * Domain Path: /languages/
+ * License:     GPLv2 or later (license.txt)
+ */
+
+// @codingStandardsChangeSetting WordPress.Utils.I18nTextDomainFixer old_text_domain debug-bar
+
+/*
+ Plugin Name: Text domain which should be fixed, multi-line comment, minimal headers.
+ Description: Adds a debug menu to the admin bar that shows query, cache, and other helpful debugging information.
+ Text Domain: debug-bar
+ */
+
+/***
+ * Not the plugin header, don't get confused with consecutive comments.
+ *
+ * When logged in as a super admin, these functions will run to provide
+ * debugging information when specific super admin menu items are selected.
+ *
+ * They are not used when a regular user is logged in.
+ */
+class Debug_Bar {
+
+	/**
+	 * Not the plugin header.
+	 *
+	 * @var string Text Domain: let's make sure not the sniff doesn't confuse docblocks.
+	 */
+	const VERSION = '0.9.1-alpha';
+}
+
+// @codingStandardsChangeSetting WordPress.Utils.I18nTextDomainFixer old_text_domain debug-bar-constants
+
+/**
+ * Text domain which should be fixed, docblock format with PHP doc tags.
+ *
+ * @package     WordPress\Plugins\Debug Bar Constants
+ * @author      Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ * @link        https://github.com/jrfnl/Debug-Bar-Constants
+ * @version     2.0.0
+ *
+ * @copyright   2013-2018 Juliette Reinders Folmer
+ * @license     http://creativecommons.org/licenses/GPL/2.0/ GNU General Public License, version 2 or higher
+ *
+ * @wordpress-plugin
+ * Plugin Name: Debug Bar Constants
+ * Plugin URI:  https://wordpress.org/plugins/debug-bar-constants/
+ * Description: Debug Bar Constants adds new panels to Debug Bar that display all the defined constants for the current request. Requires "Debug Bar" plugin.
+ * Version:     2.0.0
+ * Author:      Juliette Reinders Folmer
+ * Author URI:  http://www.adviesenzo.nl/
+ * Depends:     Debug Bar
+ * Text Domain: debug-bar-constants
+ * Domain Path: /languages
+ * Copyright:   2013-2018 Juliette Reinders Folmer
+ */
+
+// @codingStandardsChangeSetting WordPress.Utils.I18nTextDomainFixer old_text_domain a2-optimized
+
+//  Plugin Name: Text domain which should be fixed, multi-line //-style comment.
+//  Plugin URI: https://wordpress.org/plugins/a2-optimized/
+//  Version: 2.0.1
+//  Author: A2 Hosting
+//  Author URI: https://www.a2hosting.com/
+//  Description: A2 Optimized - WordPress Optimization Plugin
+//  Text Domain: a2-optimized
```

Comment:
Can one of these tests have more or less whitespace after the `Text Domain:`, just to show it is whitespace-agnostic?

Reply:
I'm not 100% sure what you mean. The varying comments in both this file as well as the `.css` test file, have varying layouts which is how this is tested.

The _adding_ of a missing header is not whitespace agnostic: it will basically take the previous line, copy it, change it and add it to the file, whitespace warts and all.

As for replacing - do you mean tabs vs spaces within the `Text Domain` line ? That is something I could possibly improve upon.
All the same, for multi-line function calls, the sniff also had to make a judgement call and if needed adds a tab indentation.

Reply:
For the reading of the existing text domain, to know whether it needs replacing or not.

All I see right now, is `Text Domain: Foo` - that is, exactly one space between the colon and the text domain.

Dropping in some variants would show it's whitespace-agnostic and possibly case insensitive (if WP supports that as well):

```
Text Domain:foo
Text Domain:     foo
Text Domain:[tab]foo
text domain: FOO
```

Reply:
IIRC correctly, `text-domains` are case-sensitive, so I don't think we need to allow for that (and the sniff contains a check that the `new_text_domain` is lowercase, dash-separated.

The last new commit I added, already contains variants of the headers with inline tabs.

This is mostly relevant for the _adding_ of a new text domain header, not so much for the replacing, as replacing doesn't touch the whitespace, it just replaces the `text-domain` string within the larger comment string.